[SDK-3532] Expo readme documentation (#507)
auth0/react-native-auth0: README.md

> For more info please read the [React Native docs](https://facebook.github.io/react-native/docs/linking.html).

#### Expo

> :warning: This SDK is not compatible with "Expo Go" app because of custom native code. It is compatible with Custom Dev Client and EAS builds

> To use the SDK with Expo, we need to configure the app at build time. We need to provide the `domain` and optionally the `customScheme` value through the [Config Plugin](https://docs.expo.dev/guides/config-plugins/) by adding the following snippet to _app.json_ or _app.config.js_

Reviewer: A suggestion, for conciseness and readability:

```suggestion
To use the SDK with Expo, configure the app at build time by providing the `domain` and (optionally) the `customScheme` values through the [Config Plugin](https://docs.expo.dev/guides/config-plugins/). To do this, add the following snippet to _app.json_ or _app.config.js_:
```

```json
{
  "expo": {
    ...
    "plugins": [
      [
        "react-native-auth0",
        {

>           "domain": string,
>           "customScheme": string (optional),

Reviewer: Wonder if it's better to use placeholder values here that we use elsewhere. e.g.:

```suggestion
          "domain": "YOUR_AUTH0_DOMAIN",
          "customScheme": "YOUR_CUSTOM_SCHEME",
```
